src/Content/ContentStorageHandler.php

```diff
@@ -158,7 +191,20 @@ public function touchLanguage(Language $language): void
 	 *
 	 * @param array<string, string> $fields Content fields
 	 *
-	 * @throws \Kirby\Exception\NotFoundException If the version does not exist
+	 * @throws \Kirby\Exception\Exception If the file cannot be written
+	 */
+	public function update(VersionId $versionId, Language $language, array $fields): void
+	{
+		$this->ensure($versionId, $language);
+		$this->write($versionId, $language, $fields);
+	}
+
+	/**
+	 * Writes the content fields of an existing version
+	 *
+	 * @param array<string, string> $fields Content fields
+	 *
+	 * @throws \Kirby\Exception\Exception If the content cannot be written
 	 */
-	abstract public function update(VersionId $versionId, Language $language, array $fields): void;
+	abstract protected function write(VersionId $versionId, Language $language, array $fields): void;
```

Comment:
This is another reason for a separate interface. A handler implementation that has separate `create()` and `update()` methods does not need `write()` but still needs to implement it. With a separate interface, this becomes a non-issue as the interface only defines the public methods.